### PR TITLE
Add head_mask and decoder_head_mask to PyTorch LED

### LINKED ISSUE #9814: Missing head_mask and decoder_head_mask arguments in encoder-decoder models
# 🚀 Feature request

Following the PRs #9569, #9634 and #9639, there are other encoder-decoder models, which either do not support `head_mask` and `decoder_head_mask` input arguments at all or can be only provided with a single `head_mask` argument used for head masking both in encoder and decoder. It would be, therefore, nice to make this feature uniform over all the decoder-models.

<hr>

**Models:**

| Model | Pytorch | TensorFlow | PR | Copy dependency |
| ------ | :------: | :---------: | :--: | :-----: |
| BERTGeneration | ☑️  | ✖️  | - | - |
| EncoderDecoderModel | ☑️  | ✖️  | - | - |
| FSMT | ☑️  | ✖️  | #9819 | - |
| LED | ☑️ | ☑️ | PT - #9856 ; TF | - |
| ProphetNet | ☑️  | ✖️  | - | - |
| Longformer | ☑️  | ✖️  | #9856  | LED |

## Your contribution

I'm happy to add this feature in the following days, both for PyTorch and TensorFlow models. (Likely in shorter PRs in order not to create large, overwhelming PRs)

<hr>

Reviewers: @patrickvonplaten, @jplu, @sgugger, @LysandreJik, @stas00 .

### PR DESCRIPTION
**Pull request by stancld**
This PR implements `head_mask` and `decoder_head_mask` for PyTorch LED (and Longformer as there's a copy dependency) and it is the follow-up to the open issue #9814. 

**Motivation:** This PR is a part of an endeavour to enable the usage of `head_mask` and `decoder_head_mask` for all encoder-decoder transformers following the recent work on BART-like models (#9569).

<hr>

Fixes: https://github.com/huggingface/transformers/issues/9814

Reviewers: @patrickvonplaten @LysandreJik @stas00

### CODE REVIEW

**Review by patrickvonplaten — APPROVED** [review]
Great! Very clean implementation, thanks for taking care of this :-)

**Review by LysandreJik — APPROVED** [review]
Fantastic! Thanks for working on that @stancld!

File: src/transformers/models/led/modeling_led.py [hunk 1]
```diff
@@ -810,6 +834,12 @@ def forward(
             attn_weights = attn_weights.view(bsz * self.num_heads, tgt_len, src_len)
 
         attn_weights = F.softmax(attn_weights, dim=-1)
+        if layer_head_mask is not None:
+            assert layer_head_mask.size() == (
+                self.num_heads,
+            ), f"Head mask for a single layer should be of size {(self.num_heads,)}, but is {layer_head_mask.size()}"
```
Comment (sgugger) [inline src/transformers/models/led/modeling_led.py:840]: Complete nit, but those asserts are not super well formatted. Can we replace them by if and raise a proper error?
```suggestion
            if layer_head_mask.size() == self.num_heads:
                raise ValueError(f"Head mask for a single layer should be of size {(self.num_heads,)}, but is {layer_head_mask.size()}")
```

(If you do one do all of them.)

**Review by sgugger — APPROVED** [review]
Very clean, thanks for your PR! Just one styling nit, but feel free to ignore.

File: src/transformers/models/led/modeling_led.py (hunk 1 above)
Comment (stas00) [reply src/transformers/models/led/modeling_led.py:840]: much better - I tried to find a way to make the autoformatter be nice, but couldn't figure out. Yours is great and it reads even better.

Comment (stancld) [reply src/transformers/models/led/modeling_led.py:840]: @sgugger Thank you very much for your suggestion, I definitely agree with this. Do you think I should create a new PR to replace this part in all the models where desired?

Comment (stas00) [reply src/transformers/models/led/modeling_led.py:840]: Since it was a nit and not a requirement, I'd say whatever works the best for you, @stancld - thank you!

Comment (stancld) [reply src/transformers/models/led/modeling_led.py:840]: @stas00 Thanks for a quick reply. As such, I would leave it now as it is and I will replace these assertions in a new PR once there will be `head_mask` and `decoder_head_mask` implemented for all encoder-decoder models. Just in order not to mix this change in this PR, even though it's only a minor one; if it's okay :)

Comment (stas00) [reply src/transformers/models/led/modeling_led.py:840]: That works well, @stancld! 

Thank you for your great contribution!
